sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -107,7 +107,65 @@ APIs for managing configurations for devices and modules.
 <details>
 
 ```csharp
+/// <summary>
+/// Gets a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> GetConfigurationAsync(string configurationId, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Creates a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configuration">Twin configuration to create</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> CreateConfigurationAsync(TwinConfiguration configuration, CancellationToken cancellationToken = default)
 
+/// <summary>
+/// Updates a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="configuration">Twin configuration to update</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> UpdateConfigurationAsync(string configurationId, TwinConfiguration configuration, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Deletes a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="precondition">The condition on which to perform this operation</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>The Http Response</returns>
+public virtual async Task<Response> DeleteConfigurationAsync(string configurationId, IfMatchPrecondition precondition = IfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
```

Comment:
should the default be UnconditionalIfMatch since it is * for default on service?

Reply:
Our SDK defaults to "IfMatch" in the devices and modules APIs for what that's worth